src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+              SuperstructureStates.GROUND_INTAKE_CUBE_PREP
+            is SuperstructureRequest.Home -> SuperstructureStates.HOME_PREP
+            is SuperstructureRequest.Idle -> SuperstructureStates.IDLE
+            is SuperstructureRequest.PrepScore -> {
+              when (usingGamePiece) {
+                GamePiece.CONE -> SuperstructureStates.SCORE_PREP
+                GamePiece.CUBE -> SuperstructureStates.SCORE_PREP
+                else -> {
+                  currentRequest = SuperstructureRequest.Idle()
+                  SuperstructureStates.IDLE
+                }
+              }
+            }
+            is SuperstructureRequest.SingleSubstationIntake -> {
+              when (usingGamePiece) {
+                GamePiece.CONE -> SuperstructureStates.SINGLE_SUBSTATION_INTAKE_CONE
+                GamePiece.CUBE -> SuperstructureStates.SINGLE_SUBSTATION_INTAKE_CUBE
+                GamePiece.NONE -> SuperstructureStates.IDLE
+              }
+            }
+            is SuperstructureRequest.Tuning -> SuperstructureStates.TUNING
+            else -> currentState
+          }
+      }
+      SuperstructureStates.HOME_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedDownAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition) {
+          nextState = SuperstructureStates.HOME
+        }
+      }
+      SuperstructureStates.HOME -> {
+        // Outputs
+        elevator.currentRequest = Request.ElevatorRequest.Home()
+        manipulator.currentRequest = Request.ManipulatorRequest.Home()
+
+        // Transition
+        if (elevator.isHomed && manipulator.isHomed) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.intakeAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.groundIntakeCubeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE -> {
```

Comment:
I think the logic for which game piece the manipulator is holding should live in these states, btw -- like if the manipulator sees a current spike, and the superstructure is in this state, we can say `holdingGamePiece = CUBE` or something like this. We would want this because the manipulator now doesn't have any idea of what game piece it may be trying to grab, because the logic for that is now in the superstructure